plugins/playlist-backend/src/service/router.ts

```diff
@@ -227,9 +227,10 @@ export async function createRouter(
       };
     });
 
-    const token = getBearerTokenFromAuthorizationHeader(
-      req.header('authorization'),
-    );
+    const { token } = await auth.getPluginRequestToken({
+      onBehalfOf: await httpAuth.credentials(req),
+      targetPluginId: 'catalog',
+    });
 
     // TODO(kuanpg): entities in this playlist that no longer exist in the catalog will be
```

Comment:
Just noticed I spelled my own username wrong 😅 - mind adding this correction as well? 
```suggestion
    // TODO(kuangp): entities in this playlist that no longer exist in the catalog will be
```